ext/vulnmdsrc/nvd/json.go

```diff
@@ -0,0 +1,184 @@
+// limitations under the License.
+
+package nvd
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type nvd struct {
+	Entries []nvdEntry `json:"CVE_Items"`
+}
+
+type nvdEntry struct {
+	CVE               nvdCVE    `json:"cve"`
+	Impact            nvdImpact `json:"impact"`
+	PublishedDateTime string    `json:"publishedDate"`
+}
+
+type nvdCVE struct {
+	Metadata nvdCVEMetadata `json:"CVE_data_meta"`
+}
+
+type nvdCVEMetadata struct {
+	CVEID string `json:"ID"`
+}
+
+type nvdImpact struct {
+	BaseMetricV2 nvdBaseMetricV2 `json:"baseMetricV2"`
+	BaseMetricV3 nvdBaseMetricV3 `json:"baseMetricV3"`
+}
+
+type nvdBaseMetricV2 struct {
+	CVSSv2 nvdCVSSv2 `json:"cvssV2"`
+}
+
+type nvdCVSSv2 struct {
+	Score            float64 `json:"baseScore"`
+	AccessVector     string  `json:"accessVector"`
+	AccessComplexity string  `json:"accessComplexity"`
+	Authentication   string  `json:"authentication"`
+	ConfImpact       string  `json:"confidentialityImpact"`
+	IntegImpact      string  `json:"integrityImpact"`
+	AvailImpact      string  `json:"availabilityImpact"`
+}
+
+type nvdBaseMetricV3 struct {
+	CVSSv3              nvdCVSSv3 `json:"cvssV3"`
+	ExploitabilityScore float64   `json:"exploitabilityScore"`
+	ImpactScore         float64   `json:"impactScore"`
+}
+
+type nvdCVSSv3 struct {
+	Score              float64 `json:"baseScore"`
+	AttackVector       string  `json:"attackVector"`
+	AttackComplexity   string  `json:"attackComplexity"`
+	PrivilegesRequired string  `json:"privilegesRequired"`
+	UserInteraction    string  `json:"userInteraction"`
+	Scope              string  `json:"scope"`
+	ConfImpact         string  `json:"confidentialityImpact"`
+	IntegImpact        string  `json:"integrityImpact"`
+	AvailImpact        string  `json:"availabilityImpact"`
+}
+
+var vectorValuesToLetters = map[string]string{
+	"NETWORK":          "N",
+	"ADJACENT_NETWORK": "A",
+	"LOCAL":            "L",
+	"HIGH":             "H",
+	"MEDIUM":           "M",
+	"LOW":              "L",
+	"NONE":             "N",
+	"SINGLE":           "S",
+	"MULTIPLE":         "M",
+	"PARTIAL":          "P",
+	"COMPLETE":         "C",
+
+	// CVSSv3 only
+	"PHYSICAL":  "P",
+	"REQUIRED":  "R",
+	"CHANGED":   "C",
+	"UNCHANGED": "U",
+}
+
+func init() {
+	vectorValuesToLetters = make(map[string]string)
+	vectorValuesToLetters["NETWORK"] = "N"
+	vectorValuesToLetters["ADJACENT_NETWORK"] = "A"
+	vectorValuesToLetters["LOCAL"] = "L"
+	vectorValuesToLetters["HIGH"] = "H"
+	vectorValuesToLetters["MEDIUM"] = "M"
+	vectorValuesToLetters["LOW"] = "L"
+	vectorValuesToLetters["NONE"] = "N"
+	vectorValuesToLetters["SINGLE"] = "S"
+	vectorValuesToLetters["MULTIPLE"] = "M"
+	vectorValuesToLetters["PARTIAL"] = "P"
+	vectorValuesToLetters["COMPLETE"] = "C"
+
+	// CVSSv3 only
+	vectorValuesToLetters["PHYSICAL"] = "P"
+	vectorValuesToLetters["REQUIRED"] = "R"
+	vectorValuesToLetters["CHANGED"] = "C"
+	vectorValuesToLetters["UNCHANGED"] = "U"
+}
```

Comment:
I couldn't figure out how to make the GitHub suggestion delete this, too. ☹️

Reply:
Whoops! I didn't notice that merging from the GitHub UI. Fixed. (Do you want any history re-written?)